pex/variables.py

```diff
@@ -332,16 +350,37 @@ def patch(self, **kw):
         present. The rest will be added to the environment or else updated if already present in
         the environment.
         """
+        disable_env = self._maybe_get_bool_var("PEX_DISABLE_VARIABLES", kw)
+
         old_environ = self._environ
         self._environ = self._environ.copy()
+        if disable_env:
+            for k in list(self._environ):
+                if k != "PEX_DISABLE_VARIABLES" and k.startswith("PEX"):
+                    self._environ.pop(k)
         for k, v in kw.items():
             if v is None:
                 self._environ.pop(k, None)
+            elif disable_env and k != "PEX_DISABLE_VARIABLES" and k.startswith("PEX"):
+                self._environ.pop(k, None)
             else:
                 self._environ[k] = v
+
         yield self._environ
         self._environ = old_environ
 
+    @defaulted_property(default=False)
+    def PEX_DISABLE_VARIABLES(self):
+        # type: () -> bool
+        """Boolean.
+
+        Disable reading of all PEX_* variables (except this one) from all sources. Both PEX_*
```

Comment:
This is the documentation I'm referring to, that specifies `PEX_` with an underscore.

Reply:
Ah, yeah. You're right, but I'll take the opposite tack and (e)VADE.

Reply:
Hrm, you're right I think. There is precedent in that file and PEXNOTTHATPEX is not really in Pex's namespace.